crypto/rand/rand_lib.c

```diff
@@ -1023,4 +1123,69 @@ int RAND_set_seed_source_type(OSSL_LIB_CTX *ctx, const char *seed,
         && random_set_string(&dgbl->seed_propq, propq);
 }
 
-#endif
+int RAND_set1_random_provider(OSSL_LIB_CTX *ctx, OSSL_PROVIDER *prov)
```

Comment:
If this is not ref counted then its a set0()?

Reply:
This is assuming that something else loads the provider (config or code) and that the random provider wont become a floating freed ptr..

Reply:
I think this point is what is holding up the PR being approved.

Reply:
Looks good to me otherwise.

Reply:
`_set1` means that ownership is not transferred and the caller is responsible for freeing.
That's what's happening here.

I find the `[sg]et[01]` stuff fairly confusing.

Reply:
> Looks good to me otherwise. Would it require a changes entry?

I think a changes.md entry should be made ...